src/main/features/claim/routes/check-and-send.ts

```diff
@@ -17,6 +23,40 @@ function getClaimAmountTotal (res: express.Response): Promise<ClaimAmountTotal>
     })
 }
 
+function getName (partyDetails: PartyDetails): string {
+  return value(partyDetails, 'name')
+}
+
+function getContactPerson (partyDetails: PartyDetails): string {
+  return value(partyDetails, 'contactPerson')
+}
+function getBusinessName (partyDetails: PartyDetails): string {
+  return value(partyDetails, 'businessName')
+}
+
+function getDateOfBirth (partyDetails: PartyDetails): DateOfBirth {
+  if (partyDetails.type === PartyType.INDIVIDUAL.value) {
+    return (partyDetails as IndividualDetails).dateOfBirth
+  } else if (partyDetails.type === PartyType.SOLE_TRADER_OR_SELF_EMPLOYED.value) {
+    return (partyDetails as SoleTraderDetails).dateOfBirth
+  } else {
+    return undefined
+  }
+}
+
+function value (partyDetails: PartyDetails, fieldName: string): string {
+  if (partyDetails.type === PartyType.INDIVIDUAL.value) {
+    return (partyDetails as IndividualDetails)[`${fieldName}`]
```

Comment:
What is the point of casting instance to specific type if then non type safe method is used to traverse through properties tree.

ACTION needed

👎

Reply:
Removed by @comanirudha